VerbalExpressions.js

```diff
@@ -1,473 +1,426 @@
+     */
+    oneOrMore() {
+        return this.add('+');
+    }
+
+    /**
+     * Match the value zero or more times
+     * @param {String} value value to find
+     * @param {Integer?} lower minimum number of times the value should be repeated
+     * @param {Integer?} upper maximum number of times the value should be repeated
+     * @return {VerbalExpression} same instace of VerbalExpression
+     */
+    multiple(value, lower, upper) {
+        // Use expression or string
+        value = value.source || VerbalExpression.sanitize(value);
+
+        this.add(`(?:${value})`);
+
+        if (lower === undefined && upper === undefined) {
+            this.add('*'); // Any number of times
+        } else if (lower !== undefined && upper === undefined) {
+            this.add(`{${lower}}`);
+        } else if (lower !== undefined && upper !== undefined) {
+            this.add(`{${lower},${upper}}`);
+        }
+
+        return this;
+    }
+
+    // Capture groups //
+
+    /**
+     * Starts a capturing group
+     * @return {VerbalExpression} same instace of VerbalExpression
+     */
+    beginCapture() {
+        // Add the end of the capture group to the suffixes temporarily so that compilation continues to work
+        this._suffixes += ')';
+        return this.add('(');
+    }
+
+    /**
+     * Ends a capturing group
+     * @return {VerbalExpression} same instace of VerbalExpression
+     */
+    endCapture() {
+        // Remove the last parenthesis from the _suffixes and add it to the regex
+        this._suffixes = this._suffixes.slice(0, -1);
+        return this.add(')');
+    }
+
+    // Miscellaneous //
+
+    /**
+     * Shorthand function for the String.replace function to allow for a more logical flow.
+     * @param {String} source string to search for
+     * @param {String} value value to replace with
+     * @return {VerbalExpression} same instace of VerbalExpression
+     */
+    replace(source, value) {
+        source = source.toString();
+        return source.replace(this, value);
+    }
+
+    /**
+     * Convert to RegExp object
+     * @return {RegExp} Converted RegExp instance
+     */
+    toRegExp() {
+        const components = this.toString().match(/\/(.*)\/([gimuy]+)?/);
+        const pattern = components[1];
+        const flags = components[2];
+
+        return new RegExp(pattern, flags);
+    }
+}
+
+// UMD (Universal Module Definition)
+// https://github.com/umdjs/umd
+((root, factory) => {
+    if (typeof define === 'function' && define.amd) {
+        // AMD. Register as an anonymous module.
+        define([], factory);
+    } else if (typeof module === 'object' && module.exports) {
+        // Node. Does not work with strict CommonJS
+        module.exports = factory();
     } else {
-        // Browser
-        root[MODULE_NAME] = createVerbalExpression;
+        // Browser globals (root is window)
+        root.VerEx = factory();
     }
-}(this));
+})(this, () => () => new VerbalExpression());
```

Comment:
Now there's a small problem with UMD.

We normally use UMD to "export" the package correctly to all environments (amd/commonjs/browser). In the browser's scope it means that we should only export the `VerEx` variable to the global scope. This is done by wrapping the whole code to an anonymous function that's executed immediately:

```
(() => { ...your code })()
```

This has an effect that any variables/classes/functions you declare inside the anonymous function will not pollute the global scope (in browser), as variables are scoped by the function.

Now you're not wrapping the `class VerbalExpression` in an anonymous function, so the class `VerbalExpression` is also injected to the global scope when using a browser environment.

I see two ways of fixing this:

1. Use some postprocessor (e.g. a Grunt plugin) to wrap the whole JS into an anonymous function
2. Do the wrapping in code, like it was done previously

Reply:
Cool. So is it fine if the postprocessor (I've done it with a grunt plugin for now but might switch to a babel plugin) wraps the transpiled code in an IIFE?

Reply:
> So is it fine if the postprocessor wraps the transpiled code in an IIFE?

Yep, sounds good :)